### PR TITLE
mkpolyline-init-locations

### PR DESCRIPTION
**Pull request by trevorphillips**
Added an init to make a polyline out of Cllocations.

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Tests/MapKitTests/MKPolylineTests.swift [hunk 1]
```diff
@@ -36,6 +44,20 @@ final class MKPolylineTests: XCTestCase {
...
 
+    func testInitWithLocations() {
+        var refCoordinates = locations.map {
+            return CLLocationCoordinate2D(latitude: $0.coordinate.latitude, longitude: $0.coordinate.longitude)
+        }
+
+        let polyline = MKPolyline(locations: locations)
+        let polyline2 = MKPolyline(coordinates: &refCoordinates, count: refCoordinates.count)
+
+        for (coordinate1, coordinate2) in zip(polyline.coordinates, polyline2.coordinates) {
```
Comment (LucianoPAlmeida) [inline Tests/MapKitTests/MKPolylineTests.swift]: Why not just use `zip(polyline.coordinates, refCoordinates)`, because the way it is being done right now we testing against the same implementation, so the result will always be the same...

File: Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift [hunk 2]
```diff
@@ -21,6 +21,16 @@ public extension MKPolyline {
         self.init(coordinates: &refCoordinates, count: refCoordinates.count)
     }
 
+    /// SwifterSwift: Create a new MKPolyline from a provided Array of locations.
+    ///
+    /// - Parameter locations: Array of CLLocation(s).
+    convenience init(locations: [CLLocation]) {
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift]: Maybe this would be nicer if we take an `[CLLocationCoordinate2D]` instead?

Comment (trevorphillips) [reply Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift]: There already exists one with that arg. I just thought it would be nice for users to have cllocation also.

File: Tests/MapKitTests/MKPolylineTests.swift (hunk 1 above)
Comment (trevorphillips) [reply Tests/MapKitTests/MKPolylineTests.swift]: i basically copied and pasted from the test above.
`    func testInitWithCoordinates() {
        var refCoordinates = coordinates

        let polyline = MKPolyline(coordinates: coordinates)
        let polyline2 = MKPolyline(coordinates: &refCoordinates, count: refCoordinates.count)

        for (coordinate1, coordinate2) in zip(polyline.coordinates, polyline2.coordinates) {
            XCTAssertEqual(coordinate1.latitude, coordinate2.latitude, accuracy: 0.000000001)
            XCTAssertEqual(coordinate1.longitude, coordinate2.longitude, accuracy: 0.000000001)
        }
    }`

File: Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift (hunk 2 above)
Comment (LucianoPAlmeida) [reply Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift]: Humm, I see now :)

File: Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift [hunk 3]
```diff
@@ -21,6 +21,16 @@ public extension MKPolyline {
...
+        var refCoordinates = locations.map {
```
Comment (guykogus) [inline Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift]: You can shorten this to 1 line
```suggestion
        var refCoordinates = locations.map { $0.coordinate }
```

File: Tests/MapKitTests/MKPolylineTests.swift [hunk 4]
```diff
@@ -36,6 +44,20 @@ final class MKPolylineTests: XCTestCase {
         }
     }
 
+    func testInitWithLocations() {
+        var refCoordinates = locations.map {
```
Comment (guykogus) [inline Tests/MapKitTests/MKPolylineTests.swift]: Same simplification
```suggestion
        var refCoordinates = locations.map { $0.coordinate }
```

File: Tests/MapKitTests/MKPolylineTests.swift (hunk 1 above)
Comment (guykogus) [reply Tests/MapKitTests/MKPolylineTests.swift]: I agree with @LucianoPAlmeida regarding re-using the implementation. I don't like that it was done above either. Just compare `polyline.coordinates` with `coordinates`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
 ## Upcoming Release
 
 ### Added
+- **MKPolyline**:
+    - Added `init(locations:)` to MKPolyline. [#805](https://github.com/SwifterSwift/SwifterSwift/pull/805) by [trevorphillips](https://github.com/trevorphillips)
```
Comment (guykogus) [inline CHANGELOG.md]:
```suggestion
  - Added `init(locations:)` to MKPolyline. [#805](https://github.com/SwifterSwift/SwifterSwift/pull/805) by [trevorphillips](https://github.com/trevorphillips)
```

File: Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift (hunk 3 above)
Comment (guykogus) [reply Sources/SwifterSwift/MapKit/MKPolylineExtensions.swift]: @trevorphillips why did you mark this as resolved? In case it's not clear, then function should look like this:
```Swift
convenience init(locations: [CLLocation]) {
    var coordinates = locations.map { $0.coordinate }
    self.init(coordinates: &coordinates, count: coordinates.count)
}
```

File: Tests/MapKitTests/MKPolylineTests.swift [hunk 6]
```diff
@@ -24,13 +24,27 @@ final class MKPolylineTests: XCTestCase {
         (37.329767, -121.885813)
         ].map(CLLocationCoordinate2D.init)
 
-    func testInitWithCoordinates() {
-        var refCoordinates = coordinates
+    let locations = [
```
Comment (guykogus) [inline Tests/MapKitTests/MKPolylineTests.swift]: You're just copying the coordinates above, which isn't necessary. I think what you should do is remove `.map(CLLocationCoordinate2D.init)` for `coordinates` above, then use that instead in `testInitWithCoordinates`. Similarly, you would then use `coordinates.map(CLLocation.init)` in `testInitWithLocations`. Does that make sense?

**Comment by LucianoPAlmeida** [comment]
@trevorphillips , there are some conflicts and unaddressed comments here, can you please check them so we can move this forward, thank you :))
